src/passes/J2CLItableMerging.cpp

```diff
@@ -275,16 +288,22 @@ struct J2CLItableMerging : public Pass {
         }
 
         // This is a struct.get that returns an itable type;
-        // Change to return the corresponding vtable type.
+        // Change to return the corresponding vtable type. If the vtable is a
+        // custom descriptor, change to a ref.get_desc instruction.
         Builder builder(*getModule());
-        replaceCurrent(builder.makeStructGet(
-          0,
-          curr->ref,
-          MemoryOrder::Unordered,
-          parent.structInfoByITableType[curr->type.getHeapType()]
-            ->javaClass.getStruct()
-            .fields[0]
-            .type));
+
+        HeapType& javaClass = parent.structInfoByITableType[curr->type.getHeapType()]->javaClass;
```

Comment:
A HeapType is represented as a single number, so no need to take a reference to it unless you're going to mutate it.
```suggestion
        HeapType javaClass = parent.structInfoByITableType[curr->type.getHeapType()]->javaClass;
```